### PR TITLE
Niganesh/ion changes commit reveal

### PR DESCRIPTION
**Pull request by nithyaganeshng**
Problem:
SDK is not compatible with ION v0.8 

Solution:
Made changes for SDK to create and resolve did using ION v0.8
1. Changes to payload
2. Changes to payload property names
3. Changes encoding in payload
4. The changes affect the database schema for identifiers, so have upgraded the version number of db

Validation:
Resolved the did created successfully against v0.8 ION

**Type of change:**
- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: sdk/src/main/java/com/microsoft/did/sdk/datasource/db/SdkDatabase.kt [hunk 1]
```diff
@@ -24,7 +24,7 @@ import com.microsoft.did.sdk.datasource.db.dao.VerifiableCredentialDao
  * More info:
  * https://developer.android.com/topic/libraries/architecture/room
  */
-@Database(entities = [VerifiableCredential::class, PortableIdentityCard::class, Identifier::class, Receipt::class], version = 1)
+@Database(entities = [VerifiableCredential::class, PortableIdentityCard::class, Identifier::class, Receipt::class], version = 2)
```
Comment (symorton) [inline sdk/src/main/java/com/microsoft/did/sdk/datasource/db/SdkDatabase.kt:27]: nice :)

File: sdk/src/main/java/com/microsoft/did/sdk/identifier/IdentifierCreator.kt [hunk 2]
```diff
@@ -43,14 +44,13 @@ class IdentifierCreator @Inject constructor(
     fun create(methodName: String): Result<Identifier> {
         val signatureKeyReference = "${SIGNATURE_KEYREFERENCE}_$methodName"
         val recoveryKeyReference = "${RECOVERY_KEYREFERENCE}_$methodName"
+        val updateKeyReference = "${UPDATE_KEYREFERENCE}_$methodName"
```
Comment (symorton) [inline sdk/src/main/java/com/microsoft/did/sdk/identifier/IdentifierCreator.kt:47]: what does method name mean again?

File: sdk/src/main/java/com/microsoft/did/sdk/identifier/IdentifierCreator.kt [hunk 3]
```diff
@@ -59,9 +59,8 @@ class IdentifierCreator @Inject constructor(
                     alias,
                     signatureKeyReference,
                     recoveryKeyReference,
-                    IDENTIFIER_SECRET_KEY_NAME,
-                    updateRevealValue,
-                    recoveryRevealValue
+                    updateKeyReference,
+                    IDENTIFIER_SECRET_KEY_NAME
```
Comment (symorton) [inline sdk/src/main/java/com/microsoft/did/sdk/identifier/IdentifierCreator.kt:63]: why is this called Identifier_Secret_Key_Name again?

File: sdk/src/main/java/com/microsoft/did/sdk/identifier/IdentifierCreator.kt (hunk 2 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/identifier/IdentifierCreator.kt:47]: This method is for creating master did. The additional key reference is for update keys whose public key will be used as commit-reveal value now just like recovery.

File: sdk/src/main/java/com/microsoft/did/sdk/identifier/IdentifierCreator.kt (hunk 3 above)
Comment (nithyaganeshng) [reply sdk/src/main/java/com/microsoft/did/sdk/identifier/IdentifierCreator.kt:63]: It is basically name for master did. This is how it was called before. Now, that we have master and pairwise dids, it makes sense to rename it to MASTER_IDENTIFIER_NAME. Do you have any other name suggestions?

**Pull request merged**